### ISSUE TITLE
Lambda: SNS events subscriptions only partially registered

### ISSUE THREAD

**Issue by beInDev** [serverless/serverless#3719]
# This is a Bug Report

## Description

For bug reports:
* What went wrong?
When I deploy a certain service, not all pre-existing SNS events subscriptions are registered in the lambda triggers. (18 out of 22)
* What did you expect should have happened?
That all the events are properly subscribed too
* What was the config you used?
```YAML
custom:
     baseEventName: { Fn::Join: [':', ["arn:aws:sns",Ref: "AWS::Region", Ref: "AWS::AccountId", "event-${opt:stage, self:provider.stage}"]] }
functions:
    onIncomingMessage:
        handler: index.onIncomingMessage
        events:
        - sns:
            arn: { Fn::Join: ['-', ["${self:custom.baseEventName}", "onBusinessUpdated"]] }
            topicName: onBusinessUpdated
        - sns:
            arn: { Fn::Join: ['-', ["${self:custom.baseEventName}", "onBusinessAccountCompleted"]] }
            topicName: onBusinessAccountCompleted
.... (20 more like that)
```
* What stacktrace or error message from your provider did you see?
N/A

According to CloudFormation, everything was created succesfully

## Additional Data

* ***Serverless Framework Version you're using***: 1.13.2 (but happens with 1.14.0 too)
* ***Operating System***: Linux Mint
* ***Stack Trace***: N/A
* ***Provider Error messages***: N/A

**Comment by pmuens** [comment]
@beInDev thanks for reporting! 👍 

Differente event source, but might be related to (https://github.com/serverless/serverless/issues/3713).

Seems like the `SNS` resource templates could add some `DependsOn` configs: https://github.com/serverless/serverless/blob/847b5385b59c5d87763c5e5eac98878cc4a065da/lib/plugins/aws/package/compile/events/sns/index.js#L148-L155

https://github.com/serverless/serverless/blob/847b5385b59c5d87763c5e5eac98878cc4a065da/lib/plugins/aws/package/compile/events/sns/index.js#L164-L172

**Comment by beInDev** [comment]
@pmuens 
Would `DependsOn` be able to reference a resource from another stack or would we have to reference it somehow?

I could just use ImportValue in this stack and export my events from the other too, I guess it could work?

**Comment by pmuens** [comment]
> Would DependsOn be able to reference a resource from another stack or would we have to reference it somehow?

🤔 not entirely sure. I tried to add some `DependsOn` statements in a nested stack to reference resource in the parent stack and that didn't worked until I passed them into the template with the help of stack parameters (as far as I remember). Something along those lines might apply here as well.

> I could just use ImportValue in this stack and export my events from the other too, I guess it could work?

Yes, that sounds like the best solution IMHO. This will also ensure that you cannot remove the other stack since then there's a dependency between those two.
